src/picongpu/include/simulationControl/MovingWindow.hpp

```diff
@@ -63,20 +63,22 @@ class MovingWindow
         /* later used to calculate smoother offsets */
         float_64 stepsInFutureAfterComma = stepsInFuture_tmp - (float_64) stepsInFuture;
 
-        /* round to nearest step so we get smaller sliding dfference
+        /* round to nearest step so we get smaller sliding difference
          * this is valid if we activate sliding window because y direction has
          * the same size for all gpus
          */
-        const uint32_t stepsPerGPU = (uint32_t) math::floor(
-                                                            (float_64) (subGrid.getLocalDomain().size.y() * cell_height) / light_way_per_step + 0.5);
-        const uint32_t firstSlideStep = stepsPerGPU * devices_y - stepsInFuture;
-        const uint32_t firstMoveStep = stepsPerGPU * (devices_y - 1) - stepsInFuture;
+        const float_64 stepsPerGPU = (float_64) (subGrid.getLocalDomain().size.y() * cell_height) / light_way_per_step;
+	const uint32_t firstSlideStep = (uint32_t) math::floor(   stepsPerGPU * (float_64) devices_y 
+                                                                - (float_64) stepsInFuture + 0.5           );
```

Comment:
offline discussion: we will remove the `+ 0.5` to avoid troubles caused by "early slides"